### ISSUE TITLE
Support Display the status of the waiting task only while waiting.

### ISSUE THREAD

**Issue by kahirokunn** [Nukesor/pueue#497]
### A detailed description of the feature you would like to see added.

Display the status of the waiting task only while waiting.

### Explain your usecase of the requested feature

I want to add a task to pueue and then WAIT to execute it in parallel, but I don't know the status of the task.

**Comment by Nukesor** [comment]
Could you be a bit more explicit about the requested feature?

I don't get it from your current explanation. A proper use-case and an example would be nice.

**Comment by Nukesor** [comment]
Since you seem to want to "WAIT" for a task ( even though the title and previous sentence emphasizes `display`), did you take a look at `pueue wait`?

**Comment by Nukesor** [comment]
I'm planning to publish a new release soonish.

Is this topic still an issue or can this be closed :)

**Comment by kahirokunn** [comment]
Yes, I’m using pueue wait.

**Comment by kahirokunn** [comment]
I will share my commands

**Comment by Nukesor** [comment]
From the sound of it, it feels like you want an interactive live-updating status interface.
There were ideas at some point to build an ncurses interface for pueue:
https://github.com/Nukesor/pueue/issues/49


I'm curious, why do you need the `pueue status` info in your script visible while waiting though?
Pueue wait shows all important information on the task's current status.

If you want additional info, you could also have a `while true; do pueue status && sleep 1 && clear; done` in another terminal.

**Comment by kahirokunn** [comment]
![CleanShot 2024-02-16 at 09 52 24@2x](https://github.com/Nukesor/pueue/assets/22343391/bb8af4ed-1d28-4ecb-aedd-bf39e49fd373)

In pueue wait we would also like to see the logs.

**Comment by kahirokunn** [comment]
We think it would be nice to see the last part of the log in pueue status as well.

Thx 🙏

**Comment by Nukesor** [comment]
I'm sorry, but I have to ask again. I still don't know your usecase, just that you would like to have this implemented.

So let me re-iterate:
- I'm curious, why do you need the pueue status info in your script visible while waiting though?
- Why do you need log output in your status command? If anything fails, just do a `pueue log` on the task you're interested in.

Regarding log output of tasks in `wait` and `status.
I don't see a way of doing this without making the UI very crowded and confusing. I would like to see some design proposals first on what you have in mind before discussing this in detail.

**Comment by kahirokunn** [comment]
The progress of the currently executed task will appear in the log.
Being able to check them gives a sense of security while waiting.

**Comment by kahirokunn** [comment]
If it is just an API, progress may not be necessary.
However, I am a human being and would like to see how the task is currently doing.

**Comment by Nukesor** [comment]
> The progress of the currently executed task will appear in the log.
   Being able to check them gives a sense of security while waiting.

I think this is definitely a good use case for the pueue ncurses tui as I mentioned before :)

To have live log output, pueue would have to take control over the terminal, as one would have to show the last lines of log output for each command that's currently running, including cleaning those logs up to keep the wait history clean.
That's too much work and functionality for a command that's mainly designed to enable simple scripting.

An interactive tui that allows you to dynamically explore tasks and their logs in a "live" way, seems to be a much better fit, as such a tool would also be easily extensible for further requested features.

I would rather like to build a proper tui from the get-go, instead of patching `wait` and the other commands until they become a tui by accident.

> If it is just an API, progress may not be necessary.

There's sadly no universal way to track progress of tasks, as tasks can contain arbitrary bash and programs don't have a unified way of reporting any kind of progress.

**Comment by kahirokunn** [comment]
What is pueue ncurses tui?
I couldn't find such an option in `puepe` cli. 👀

**Comment by Nukesor** [comment]
It doesn't exist yet :).
It was a previously requested feature and would need to be built first.

**Comment by kahirokunn** [comment]
Okay. Thx.

**Comment by Nukesor** [comment]
I'm closing this for now due to missing specification.

If somebody comes up with a nice design on how to build a interactive TUI, feel free to create a new ticket :)
